src/main.ts

```diff
@@ -104,18 +108,30 @@ const tryRedirect = async () => {
     throw new Error('Unexpected error, network config undefined');
   }
 
-  const rpc = networkConfig.rpc;
-  const banner = networkConfig.banner;
-  const timeoutDurationMS = 10_000;
+  const { apiAddrs, banner } = networkConfig;
   let redirectUrl: URL | undefined = undefined;
+  let didFail = false;
+
+  setTimeout(() => {
+    if (!didFail && !redirectUrl) {
+      showError('Connection is taking longer than expected. Please wait.');
+    }
+  }, timeoutDurationMS);
 
   try {
-    redirectUrl = await (Promise.race([
-      getRedirectUrl(rpc),
-      new Promise((_, rej) => setTimeout(rej, timeoutDurationMS)),
-    ]) as Promise<URL>);
+    redirectUrl = await getRedirectUrl(apiAddrs);
   } catch {
-    showError('No endorsed UI found.');
+    // Retry once as specified in https://github.com/Agoric/ui-kit/issues/56
```

Comment:
```suggestion
    // Retry all nodes once more as specified in https://github.com/Agoric/ui-kit/issues/56
```

Reply:
Moved the retry inside vstorage-query.ts where it's more self-explanatory